src/main/java/goodgenerator/blocks/tileEntity/FuelRefineFactory.java

```diff
@@ -211,64 +196,37 @@ public int getTier() {
-        ArrayList<ItemStack> tItems = getStoredInputs();
-        MyRecipeAdder.NaqFuelRefineMapper tRecipes = MyRecipeAdder.instance.FRF;
-
-        for (int i = 0; i < tFluids.size() - 1; i++) {
-            for (int j = i + 1; j < tFluids.size(); j++) {
-                if (GT_Utility.areFluidsEqual(tFluids.get(i), tFluids.get(j))) {
-                    if ((tFluids.get(i)).amount >= (tFluids.get(j)).amount) {
-                        tFluids.remove(j--);
-                    } else {
-                        tFluids.remove(i--);
-                        break;
-                    }
-                }
-            }
-        }
+    public GT_Recipe.GT_Recipe_Map getRecipeMap() {
+        return MyRecipeAdder.instance.FRF;
+    }
 
-        for (int i = 0; i < tItems.size() - 1; i++) {
-            for (int j = i + 1; j < tItems.size(); j++) {
-                if (GT_Utility.areStacksEqual(tItems.get(i), tItems.get(j))) {
-                    if ((tItems.get(i)).stackSize >= (tItems.get(j)).stackSize) {
-                        tItems.remove(j--);
-                    } else {
-                        tItems.remove(i--);
-                        break;
-                    }
+    @Override
+    protected ProcessingLogic createProcessingLogic() {
+        return new ProcessingLogic() {
+
+            @NotNull
+            @Override
+            protected CheckRecipeResult validateRecipe(@NotNull GT_Recipe recipe) {
+                if (recipe.mSpecialValue > Tier) {
+                    return CheckRecipeResultRegistry.insufficientMachineTier(recipe.mSpecialValue);
                 }
+                return CheckRecipeResultRegistry.SUCCESSFUL;
             }
-        }
 
-        FluidStack[] inFluids = tFluids.toArray(new FluidStack[0]);
-        ItemStack[] inItems = tItems.toArray(new ItemStack[0]);
-        this.mEfficiency = 10000;
-
-        long tPower = getMaxInputEnergy_EM();
-        byte tTier = (byte) Math.max(1, GT_Utility.getTier(tPower));
-        GT_Recipe recipe = tRecipes.findRecipe(this.getBaseMetaTileEntity(), false, V[tTier], inFluids, inItems);
-        if (recipe != null) {
-            if (recipe.mSpecialValue > Tier) return false;
-            if (recipe.isRecipeInputEqual(true, inFluids, inItems)) {
-                mEUt = recipe.mEUt * (1 << (Tier - recipe.mSpecialValue));
-                mEUt = -Math.abs(mEUt);
-                mMaxProgresstime = recipe.mDuration / (1 << (Tier - recipe.mSpecialValue));
-                this.mOutputFluids = recipe.mFluidOutputs;
-                this.updateSlots();
-                return true;
+            @NotNull
+            @Override
+            protected GT_OverclockCalculator createOverclockCalculator(@NotNull GT_Recipe recipe,
+                    @NotNull GT_ParallelHelper helper) {
+                return super.createOverclockCalculator(recipe, helper)
+                        .setSpeedBoost(1F / (float) (1 << (Tier - recipe.mSpecialValue)));
```

Comment:
`mSpeedBoost` doesn't have penalty for EUt, so this line is not ported? `mEUt = recipe.mEUt * (1 << (Tier - recipe.mSpecialValue));`